packages/ui/src/components/va-counter/VaCounter.vue

```diff
@@ -0,0 +1,391 @@
+  ['change'],
+)
+
+const { createEmits: createFieldEmits, createListeners: createFieldListeners } = useEmitProxy([
+  { listen: 'click-prepend', emit: 'click:decrease-button' },
+  { listen: 'click-append', emit: 'click:increase-button' },
+  { listen: 'click-prepend-inner', emit: 'click:decrease-icon' },
+  { listen: 'click-append-inner', emit: 'click:increase-icon' },
+])
+
+export default defineComponent({
+  name: 'VaCounter',
+
+  components: { VaInputWrapper, VaIcon, VaButton },
+
+  props: {
+    ...useFormProps,
+    ...useStatefulProps,
+    // input
+    modelValue: { type: [String, Number] as PropType<string | number>, default: 0 },
+    manualInput: { type: Boolean as PropType<boolean>, default: false },
+    stateful: { type: Boolean as PropType<boolean>, default: false },
+    min: { type: Number as PropType<number>, default: undefined },
+    max: { type: Number as PropType<number>, default: undefined },
+    step: { type: Number as PropType<number>, default: 1 },
+    label: { type: String as PropType<string>, default: '' },
+    // hint
+    messages: { type: [Array, String] as PropType<string[] | string>, default: () => [] },
+    // style
+    width: { type: [String, Number] as PropType<string | number>, default: '160px' },
+    color: { type: String as PropType<string>, default: 'primary' },
+    outline: { type: Boolean as PropType<boolean> },
+    bordered: { type: Boolean as PropType<boolean> },
+    // icons & buttons
+    increaseIcon: { type: String as PropType<string>, default: 'add' },
+    decreaseIcon: { type: String as PropType<string>, default: 'remove' },
+    buttons: { type: Boolean as PropType<boolean>, default: false },
+    flat: { type: Boolean as PropType<boolean>, default: true },
+    rounded: { type: Boolean as PropType<boolean>, default: false },
+    margins: { type: [String, Number] as PropType<string | number>, default: '4px' },
+    textColor: { type: String as PropType<string | undefined>, default: undefined },
+  },
+
+  emits: [
+    'update:modelValue',
+    ...createInputEmits(),
+    ...createFieldEmits(),
+    ...useFocusEmits,
+  ],
+
+  inheritAttrs: false,
+
+  setup (props, { emit, attrs }) {
+    const input = ref<HTMLInputElement | HTMLDivElement | undefined>()
+
+    const {
+      isFocused,
+      // will be useful when we resolve problem with 'withConfigTransport'
+      focus,
+      blur,
+    } = useFocus(input, emit)
+
+    const { valueComputed } = useStateful(props, emit)
+
+    const setCountInput = ({ target } : { target: HTMLInputElement }) => {
+      valueComputed.value = Number(target?.value)
+    }
+
+    const setCountChange = ({ target } : { target: HTMLInputElement }) => {
+      calculateCounterValue(Number(target?.value))
+    }
+
+    const getRoundDownWithStep = (value: number) => {
+      if (!props.min || !props.step) { return value }
+      return props.min + props.step * Math.floor((value - props.min) / props.step)
```

Comment:
Do we really need props.min here?

```suggestion
      return props.step * Math.floor((value) / props.step)
```

Reply:
Definitely we need this. props.min is used as offset here. So if 1 is min, then all values can be only 1 + step * n.

Reply:
Yes. If the user enters a value manually, then we must round it to the nearest valid value, taking into account the initial value (`min`) and the step size (`step`).
I have added comments to the code